## SalesForce Training
Studi kasus: Perusahaan X bermaksud meningkatkan penjualan produk. Berdasarkan data penjualan sebelumnya, diketahui penjualan rata-rata sebesar $100 per transaksi.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

**Create DataFrame**
<br>
Perusahaan melakukan training kepada para pekerja sales dan diperoleh data penjualan terbaru (dari 25 sampel pekerja sales) sebagai berikut

In [2]:
# Create DataFrame
df = pd.DataFrame({"Transaction_amount" : [
    100, 150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90, 120, 100, 110, 75, 65
]})

df.head()

Transaction_amount
0                 100
1                 150
2                  50
3                 100
4                 130

In [3]:
df.tail()

Transaction_amount
20                 120
21                 100
22                 110
23                  75
24                  65

### Measure of Central Tendency Analysis

In [4]:
transaction_amount = df["Transaction_amount"]

# Central Tendency
print("Mean =", transaction_amount.mean())
print("Median =", transaction_amount.median())
print("Mode =", transaction_amount.mode()[0])

Mean = 102.6
Median = 100.0
Mode = 100


### Variability Analysis

In [5]:
# Variability
print("Variance =", np.var(transaction_amount))
print("Standard Deviation =", np.std(transaction_amount))
print("Minimum =", np.min(transaction_amount))
print("Maximum =", np.max(transaction_amount))
print("Range =", (np.max(transaction_amount) - np.min(transaction_amount)))
print("Quartile 1 =", np.quantile(transaction_amount, 0.25))
print("Quartile 2 =", np.quantile(transaction_amount, 0.5))
print("Quartile 3 =", np.quantile(transaction_amount, 0.75))

Variance = 972.2399999999997
Standard Deviation = 31.180763300471007
Minimum = 50
Maximum = 150
Range = 100
Quartile 1 = 85.0
Quartile 2 = 100.0
Quartile 3 = 120.0


Dari hasil analisis variability di atas, diperoleh nilai variansi data yang cukup besar (nilai variance mendekati 1000). Diperoleh juga nilai quartil 1 = 85 dan quartil 3 = 120 yang menunjukkan bahwa sebagian besar nilai data terkonsentrasi dalam rentang nilai 85 hingga 120. 

### Define Hypotesis H0 and H1, and alpha
* **H0**: Penjualan rata-rata = 100 per transaksi setelah dilakukan training kepada pekerja sales
* **H1**: Penjualan rata-rata > 100 per transaksi setelah dilakukan training kepada pekerja sales
* **alpha**: 5%

In [6]:
alpha = 0.05

# Calculate T-test and P-value
t_test, p_value = stats.ttest_1samp(a=transaction_amount, popmean=100)
print("t_test =", t_test)
print("P-value =", p_value)

t_test = 0.4085001556802841
P-value = 0.6865284813438117


Dari hasil nilai T-test dan P-value yang diperoleh, nilai P-value = 0,68 lebih besar daripada nilai alpha = 5% (0,05). Sehingga, gagal menolak H0: Training terhadap pekerja sales tidak memberikan pengaruh berarti, rata-rata jumlah transaksi masih di sekitaran 100.

Menentukan Critical Region

In [7]:
print("Critical Value =", stats.t.ppf(1-alpha, 24))

Critical Value = 1.7108820799094275


Dari hasil pengujian t, diperoleh bahwa nilai T-test berada di luar nilai Critical (Critical Region : t > 1.71), sehingga gagal menolak H0

**Kesimpulan**: Training pada pekerja sales belum mampu meningkatkan penjualan rata-rata, rata-rata transaksi penjualan masih di sekitaran 100